### PR TITLE
ENH: Adds support for the nsfw tag

### PR DESCRIPTION
**Pull request by SantiagoTorres**
A super-minor improvement to start the development here again...

NSFW images are hosted on imgur, we (most likely) do not want to have them as backgrounds. I added another configuration flag to the imgur fetcher to throw away images with this flag.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +75,12 @@ or description of the image. Imgur can host some really nasty things, so
 this seems to be a required feature.
 
 
+#### nsfw
+
+By default, files marked as nsfw are ignored, but you can change this by
```
Comment (ElDeveloper) [inline README.md]: Optional but add a link to http://www.urbandictionary.com/define.php?term=NSFW&defid=248838 ? haha.

File: src/bg_daemon/settings.json [hunk 2]
```diff
@@ -9,12 +9,13 @@
...
                     ],
-        
+
         "subreddits": [ "earthporn"
                     ],
 
         "blacklist_words":["gore"],
-        "mode":"recent"
+        "mode":"recent",
+        "nsfw":True
```
Comment (ElDeveloper) [inline src/bg_daemon/settings.json]: So by default we accept NSFW images?

File: src/bg_daemon/fetchers/imgurfetcher.py [hunk 3]
```diff
@@ -50,6 +50,8 @@ class imgurfetcher:
                   and randomly select an image in it. If mode is not either,
                   it will default to recent.
 
+            nsfw: tells whether accept images with the nsfw tag set
```
Comment (ElDeveloper) [inline src/bg_daemon/fetchers/imgurfetcher.py]: `whether accept` -> `whether or not to accept`

File: tests/test_imgurfetcher.py [hunk 4]
```diff
@@ -57,26 +59,31 @@ def setUp(self):
         self.bad_image_height = imgurpython.helpers.GalleryImage(link=None,
                 title=self.fetcher.keywords[0],
                 description=self.fetcher.keywords[0],
-                width=10000, height=1)
+                width=10000, height=1, nsfw=False)
```
Comment (ElDeveloper) [inline tests/test_imgurfetcher.py:62]: Maybe worth adding a test with nsfw True?

**Comment by ElDeveloper** [comment]
Looks good, just a question, how is this gonna work with my installation, will I have to add `nsfw` to my settings.json?

File: tests/test_imgurfetcher.py (hunk 4 above)
Comment (SantiagoTorres) [inline tests/test_imgurfetcher.py:62]: There is, on line 220

**Comment by SantiagoTorres** [comment]
@ElDeveloper Nope, if the setting is not set, it is defaulted to False (even if it's not in the settings.json), so it filters them automatically. If you want to see nsfw images in your settings.json, you should set it as such (that's why I made that mistake).

I think I'm ready for another round :+1:

File: tests/test_imgurfetcher.py (hunk 4 above)
Comment (ElDeveloper) [inline tests/test_imgurfetcher.py:62]: :boom:

**Comment by ElDeveloper** [comment]
Thanks @SantiagoTorres

**Pull request merged**